lib/spack/spack/test/config.py

```diff
@@ -1252,3 +1252,62 @@ def test_user_cache_path_is_default_when_env_var_is_empty(working_env):
     os.environ['SPACK_USER_CACHE_PATH'] = ''
     assert os.path.expanduser("~%s.spack" % os.sep) == \
         spack.paths._get_user_cache_path()
+
+
+github_url = "https://github.com/spack/spack/{0}/develop"
+gitlab_url = "https://gitlab.spack.io/user/repo/-/blob/config/defaults"
```

Comment:
In general I have a concern about using real URLs. I wonder if this could be refactored in the following manner:

* `spack.util.web` has a function for reading a single file from a url (this is not different from how the PR works)
* `spack.util.web` also has a function for reading a list of files from a URL that represents a directory (that would be new)
* `spack.config.fetch_remote_configs` can make use of these files
* The tests can monkeypatch the `spack.util.web` methods and automatically populate a mock config directory as needed

It might be that the tests related to verifying correct behavior of `spack.util.web` would be best done with real URLs (though I see that `spack/test/web.py` generally makes do with `file://` or mocking) but I think it would be best to isolate that to a specific, low-level function.

Reply:
🤔

Reply:
I did the refactoring but used fake URLs.